.github/workflows/tests.yml

```diff
@@ -0,0 +1,52 @@
+name: tests
+
+on:
+  push:
+    branches:
+      - master
+      - '*.x'
+  pull_request:
+  schedule:
+    - cron: '0 0 * * *'
+
+jobs:
+  tests:
+    runs-on: ubuntu-22.04
+
+    strategy:
+      fail-fast: true
+      matrix:
+        php: [7.4, '8.0', 8.1, 8.2]
+
+    name: PHP ${{ matrix.php }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php }}
+          extensions: dom, curl, libxml, mbstring, redis, zip
+          ini-values: error_reporting=E_ALL
+          tools: composer:v2
+          coverage: none
+
+      - name: Install dependencies
+        run: composer install --prefer-dist --no-interaction --no-progress
+
+      - name: Execute Psalm
+        run: vendor/bin/psalm
```

Comment:
Running Psalm first because it has a lower lifecycle to run its checks than PHPUnit.

Reply:
Wouldn't it make sense to move Psalm into a separate workflow? I would want to run PHPUnit even if Psalm fails. Additionally, both tools can perfectly run in parallel.

Reply:
We can definitely do that.

Reply:
Done.